pallets/pallet-bank/src/tests.rs

```diff
@@ -219,13 +159,170 @@ fn can_reaped() {
+            assert_ok!(Bank::stake_funds(RuntimeOrigin::signed(ALICE), 900));
+            System::reset_events();
+            assert_ok!(Bank::lock_funds_auditor(
+                RuntimeOrigin::signed(charlie),
+                ALICE,
+                200,
+                20
+            ));
+            assert_eq!(
+                System::events()[0].event,
+                RuntimeEvent::Bank(Event::<Runtime>::AuditorLock {
+                    user: ALICE.clone(),
+                    amount: 200,
+                    length: 20
+                })
+            );
+            assert_eq!(
+                Accounts::<Runtime>::get(&ALICE),
+                AccountData {
+                    free: 0,
+                    reserved: 800,
+                    locked: vec![LockedFund {
+                        id: 1,
+                        amount: 200,
+                        reason: LockReason::Auditor,
+                    }]
+                }
+            );
+
+            assert!(Bank::check_total_issuance());
+
+            assert_noop!(
+                Bank::lock_funds_auditor(RuntimeOrigin::signed(charlie), ALICE, 801, 20),
+                Error::<Runtime>::InsufficientBalance
+            );
+        });
+}
+
+#[test]
+fn can_auditor_unlock_funds() {
+    MockGenesisConfig::with_balances(vec![(ALICE, 1_000)])
+        .build()
+        .execute_with(|| {
+            let charlie: AccountId = 3u32;
+            assert_eq!(Bank::accounts(&charlie), AccountData::default());
+            assert_ok!(Roles::register_role(&charlie, Role::Auditor));
+            assert_ok!(Bank::stake_funds(RuntimeOrigin::signed(ALICE), 900));
+            
+            assert_ok!(Bank::lock_funds_auditor(
+                RuntimeOrigin::signed(charlie),
+                ALICE,
+                200,
+                20
+            ));
+            
+            assert_eq!(
+                Accounts::<Runtime>::get(&ALICE),
+                AccountData {
+                    free: 0,
+                    reserved: 800,
+                    locked: vec![LockedFund {
+                        id: 1,
+                        amount: 200,
+                        reason: LockReason::Auditor,
+                    }]
+                }
+            );
+
+            System::reset_events();
+            assert_ok!(Bank::unlock_funds_auditor(RuntimeOrigin::signed(charlie),
+            ALICE, 1));
+            assert_eq!(
+                System::events()[0].event,
+                RuntimeEvent::Bank(Event::<Runtime>::AuditorUnlock {
+                    user: ALICE.clone(),
+                    amount: 200
+                })
+            );
+
+            assert_eq!(
+                Accounts::<Runtime>::get(&ALICE),
+                AccountData {
+                    free: 200,
+                    reserved: 800,
+                    locked: vec![]
+                }
+            );
+            assert!(Bank::check_total_issuance());
+        });
```

Comment:
test some edge cases:
* other auditor can unlock.
* Can unlock funds that doesn't exist (doesn't crash)
* Can't unlock Redeem locks.